pandas/core/frame.py

```diff
@@ -3533,12 +3533,23 @@ def _combine_match_columns(self, other, func, level=None, fill_value=None):
                                    axes=[left.columns, self.index])
         return self._constructor(new_data)
 
-    def _combine_const(self, other, func, raise_on_error=True):
+    def _combine_const(self, other, func,
+                       raise_on_error=True, fill_value=None):
         if self.empty:
             return self
 
-        new_data = self._data.eval(func=func, other=other,
-                                   raise_on_error=raise_on_error)
+        if fill_value is not None:
```

Comment:
I have modified the API, `DataFrame._combine_const`, is it OK?
if yes, it seems that we need to modify `_combine_const` method of sparse and series and their caller.